src/App.tsx

```diff
@@ -1,60 +1,162 @@
+    const checkResult = Boolean(todoTitle.replace(/ /g, '').length);
+
+    if (!checkResult) {
+      setTodoTitle('');
+    }
+
+    return checkResult;
+  };
+
+  const sumbitAcces = () => {
+    return userId && emptyStringCheck();
+  };
+
+  const findLastIndex = (prevTodos: Todo[]): number => (
+    prevTodos.reduce((maxID, currentTodo) => (
+      Math.max(maxID, currentTodo.id)), 0)
+  );
+
+  const checkInputString = (str: string): string => {
+    str.trim().replace(/[^a-zA-Z\u0400-\u04FF\0-9 ]/g, '');
+
+    return str;
+  };
+
+  const updateTodos = () => {
+    setTodos((prevTodos) => (
+      [...prevTodos,
+        {
+          id: findLastIndex(prevTodos) + 1,
+          title: todoTitle,
+          userId,
+          completed: false,
+          user: getUser(userId),
+        }]
+    ));
+  };
+
+  const resetForm = () => {
+    setUserId(0);
+    setTodoTitle('');
+    setVisibleAttention(false);
+  };
+
+  const handleSubmitData = () => {
+    const acces = sumbitAcces();
+
+    if (acces) {
+      updateTodos();
+      resetForm();
+    } else {
+      setVisibleAttention(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          handleSubmitData();
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title_id">
+            Title:
+          </label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            name="title"
+            value={todoTitle}
+            id="title_id"
+            placeholder="Enter a title"
+            onChange={(event) => {
+              const inputTitle = checkInputString(event.target.value);
+
+              setTodoTitle(inputTitle);
+            }}
+          />
+          {(!todoTitle && visibleAttention) && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="user_id">
+            User:
+          </label>
+          <select
+            data-cy="userSelect"
+            name="user"
+            value={userId}
+            id="user_id"
+            onChange={(event) => {
+              const { value } = event.target;
+
+              setUserId(+value);
+            }}
+          >
             <option value="0" disabled>Choose a user</option>
+            {usersFromServer.map(user => (
+
+              <option
+                key={user.id}
+                value={user.id}
+              >
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
```

Comment:
- before creating a todo, check if a `user` was selected; if not, show an error message next to the `select` (`Please choose a user`);

Same here, you should check user id, and then use conditional rendering to show this error.

`{hasValidId && <span className="error">Please choose a user</span>}`